### PR TITLE
executor: track memory usage for merge join

### PR DESCRIPTION
**Pull request by zz-jason**

### CODE REVIEW

**Comment by XuHuaiyu** [comment]
LGTM

File: executor/merge_join.go [hunk 1]
```diff
@@ -165,18 +176,25 @@ func (t *mergeJoinInnerTable) reallocReaderResult() {
...
 	if err := e.baseExecutor.Open(ctx); err != nil {
 		return errors.Trace(err)
 	}
+
 	e.prepared = false
+	e.memTracker = memory.NewTracker(e.id, e.ctx.GetSessionVars().MemQuotaMergeJoin)
+	e.memTracker.AttachTo(e.ctx.GetSessionVars().StmtCtx.MemTracker)
+
+	e.innerTable.memTracker = memory.NewTracker("innerTable", -1)
+	e.innerTable.memTracker.AttachTo(e.memTracker)
```
Comment (zimulala) [inline executor/merge_join.go]: Do we need to detach `e.innerTable.memTracker`?

Comment (coocood) [reply executor/merge_join.go]: No, `innerTable.memTracker` is a child of `e.memTracker`, `e.memTracker` is detached in `Close`.

**Pull request merged**